opm/parser/eclipse/EclipseState/Schedule/Connection.hpp

```diff
@@ -49,62 +45,39 @@ namespace Opm {
                    const Value<double>& diameter,
                    const Value<double>& skinFactor,
                    const int satTableId,
-                   const WellCompletion::DirectionEnum direction = WellCompletion::DirectionEnum::Z);
+                   const WellCompletion::DirectionEnum direction);
 
```

Comment:
Is direction completion specific? Rename to WellConnection::DirectionEnum? 
You also have WellCompletion::StateEnum which belongs to the completion. So rename is probably not sufficient.